lib/Auth/Process/UpdateUserExtSource.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+class sspmod_perun_Auth_Process_UpdateUserExtSource extends SimpleSAML_Auth_ProcessingFilter
+{
+	private $attrMap;
+	private $attrsToConversion;
+	private $adapter;
+	const UES_ATTR_NMS = 'urn:perun:ues:attribute-def:def:';
+
+	public function __construct($config, $reserved)
+	{
+		parent::__construct($config, $reserved);
+
+		assert('is_array($config)');
+
+		if (!isset($config['attrMap'])) {
+			throw new SimpleSAML_Error_Exception("perun:UpdateUserExtSource: missing mandatory configuration option 'attrMap'.");
+		}
+
+		if (isset($config['arrayToStringConversion'])) {
+			$this->attrsToConversion = (array) $config['arrayToStringConversion'];
+		} else {
+			$this->attrsToConversion = array();
+		}
+
+		$this->attrMap = (array)$config['attrMap'];
+		$this->adapter = sspmod_perun_Adapter::getInstance(sspmod_perun_Adapter::RPC);
+	}
+
+	public function process(&$request)
+	{
+		assert('is_array($request)');
+		try {
+			$userExtSource = $this->adapter->getUserExtSource($request['Attributes']['sourceIdPEntityID'][0], $request['Attributes']['sourceIdPEppn'][0]);
+			if (is_null($userExtSource)) {
+				throw new SimpleSAML_Error_Exception("sspmod_perun_Auth_Process_UpdateUserExtSource - userExtSource is null");
```

Comment:
Please use another Exception message - for example: There is no UserExtSource with ExtSource  ... and Login ...